sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -101,6 +179,10 @@ def test_feature_importances():
     mask = np.abs(transformer.estimator_.coef_) > 1e-5
     assert_array_equal(X_new, X[:, mask])
 
+    # Test max_n_features parameter using various values
+    check_max_n_features(est, X.copy(), y)
```

Comment:
Values of `X` are being changed [here](https://github.com/qmaruf/scikit-learn/blob/ad0a8b3bc1adb0c0975558977e729d3faefafdee/sklearn/feature_selection/tests/test_from_model.py#L119) inside `check_max_n_features`. Without using `copy`, `X` is also changed outside of that function. That's why I have used `copy`.